extension.js

```diff
@@ -314,7 +318,40 @@ function workarea_for_window(win) {
 
 function target_rect_for_workarea(workarea) {
     const target_rect = workarea.copy();
```

Comment:
Remove this line and simply use `workarea.width`, `workarea.height`, etc. if you recompute all coordinates/dimensions anyway. And then `return new Meta.Rectangle({x, y, width, height});`